src/commands/deploy/utils.ts

```diff
@@ -143,3 +152,95 @@ export async function processYamlEnvironmentVariables(
 
     return newEnvObject;
 }
+
+export async function handleEnvVars(
```

Comment:
I think this function name is too generic. Without context, I can't tell what it does :))

```suggestion
export async function readAndUploadProjectEnvironment(
```

This is a suggestion, you can replace this with anything else if it sounds better/is shorter.